addons/mrp/tests/test_order.py

```diff
@@ -677,6 +678,50 @@ def test_product_produce_5(self):
         self.assertEqual(mo.move_raw_ids.filtered(lambda m: m.product_id == p1).quantity_done, 20, 'Update the produce quantity should not impact already produced quantity.')
         mo.button_mark_done()
 
+    def test_product_produce_6(self):
+        """ Plan 5 finished products, reserve and produce only 3.
+        Post the current production. Then edit an ongoing line and set consumed
+        qty and try to update the quantity on the MO to 3 and close it.
+        """
```

Comment:
```
Plan 5 finished products, reserve and produce 3. Post the current production.
Simulate an unlock and edit and, on the opened moves, set the consumed quantity
to 3. Now, try to update the quantity to produce to 3. It should fail since there
are consumed quantities. Unlock and edit, remove the consumed quantities and
update the quantity to produce to 3.
```